### PR TITLE
gh-121404: compiler_visit_* --> codegen_visit_*

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-121404

### CODE REVIEW

File: Python/compile.c [hunk 1]
```diff
@@ -1850,9 +1850,7 @@ codegen_kwonlydefaults(struct compiler *c, location loc,
                 goto error;
             }
             ADDOP_LOAD_CONST_NEW(c, loc, mangled);
-            if (compiler_visit_expr(c, default_) < 0) {
-                goto error;
-            }
+            RETURN_IF_ERROR(codegen_visit_expr(c, default_));
```
Comment (markshannon) [inline Python/compile.c]: This looks like it leaks `keys`, as it skips the `Py_XDECREF(keys);` below.
But appears `keys` is unused in this function. Maybe remove it?

File: Python/compile.c [hunk 2]
```diff
@@ -6955,9 +6953,7 @@ codegen_pattern_mapping(struct compiler *c, pattern_ty p,
             compiler_error(c, LOC(p), e);
             goto error;
         }
-        if (compiler_visit_expr(c, key) < 0) {
-            goto error;
-        }
+        RETURN_IF_ERROR(codegen_visit_expr(c, key));
```
Comment (markshannon) [inline Python/compile.c]: I think this leaks `seen`.

**Review by markshannon — COMMENTED** [review]
There is a refleak (I think)

File: Python/compile.c (hunk 2 above)
Comment (iritkatriel) [reply Python/compile.c]: ah yes

**Review by markshannon — APPROVED** [review]
Looks good

**Pull request merged**